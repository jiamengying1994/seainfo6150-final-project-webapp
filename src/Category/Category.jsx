import React from 'react';
import { Link } from 'react-router-dom';
import style from './Category.module.css';

const Category = ({ categories }) => {
  return (<div><ul className={style.categories}>
    {
       categories.map(category => {
          return (<li key={category.id}>
            <Link to={`/products/${category.category}`} className={style.link}>
              <div className={style.category}>
                <span>{category.category}</span>
              </div>
            </Link>
          </li>);
      })
    }</ul></div>)
};


export default Category;