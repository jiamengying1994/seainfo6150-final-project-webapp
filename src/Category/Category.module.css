.categories{
    display: flex;
    flex-direction: row;
    justify-content: center;
    flex-wrap: nowrap;
    margin: 1% 1% 1% 1%;
    list-style: none;
    min-width: 30rem;

}

.category{
    width: 70%;
    display: flex;
    margin:1rem;
    align-items: center;
    opacity: 0.6;
    font-size: 1.6rem;
    white-space: nowrap;

}
.category:hover{
    -webkit-transition-delay: 0.1s;
    transition-delay: 0.1s;
    opacity: 1;
}

.link {
    text-decoration: none;
    color: #333333;
}

@media (max-width: 40rem) {
    .categories{
        display: flex;
        flex-direction: row;
        justify-content: center;
        flex-wrap: wrap;
        margin: 1% 1% 1% 1%;
        list-style: none;
        min-width: 30rem;
    }

}

@media (max-width: 60rem) {
    .categories{
        display: flex;
        flex-direction: row;
        justify-content: center;
        flex-wrap: wrap;
        margin: 1% 1% 1% 1%;
        list-style: none;
        min-width: 30rem;
    }

}
