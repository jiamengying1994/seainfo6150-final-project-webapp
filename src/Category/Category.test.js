import React from "react";
import { render } from "@testing-library/react";
import Category from "./Category.jsx";
import { BrowserRouter, Route, Link } from 'react-router-dom';

describe("Category test", () => {
  it("renders correctly", () => {
    const al = [{"title": "1","image": "1"},{"title": "2","image": "2"}]
    const { container} = render (
    <BrowserRouter>
      <Category categories={al}/>
    </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});