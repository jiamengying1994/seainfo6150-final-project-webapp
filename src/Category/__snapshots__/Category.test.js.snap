// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category test renders correctly 1`] = `
<div>
  <div>
    <ul
      class="categories"
    >
      <li>
        <a
          class="link"
          href="/products/undefined"
        >
          <div
            class="category"
          >
            <span />
          </div>
        </a>
      </li>
      <li>
        <a
          class="link"
          href="/products/undefined"
        >
          <div
            class="category"
          >
            <span />
          </div>
        </a>
      </li>
    </ul>
  </div>
</div>
`;
